### PR TITLE
fix(Timeline): make reactions works

### PR DESCRIPTION
**Pull request by PascalPrecht**
This commit makes reactions in the status timeline work.
There are two things prior to this commit that are broken:

1. The logic that opens the reaction context menu always expects
   and instance of `chatsView` because it tries to calculate a users
   nickname. Such an instance isn't always available in that context, so
   the nickname logic has been moved to `appMain` for now, removing that
   dependency and therefore making it work in both, the chat view as well
   as the status view.
2. While 1) makes the context menu work, it turns out that adding and
   removing reactions inside the status timeline is still not working.
   The reason for that is, that the reactions component maintains its own
   `messageList`, which isn't aware of the fact that reactions for messages
   coming from chats of `ChatType.Profile`, need to go into a dedicated
   message list for `ChatType.Timeline`.

In other words, reactions are sent and removed from message in messagelists
that don't actually exist.

This commit fixes both of these things by ensuring the message lists
maintained by reactions are timeline aware. Also ensuring updates are
done correctly.

### CODE REVIEW

File: src/app/chat/view.nim [hunk 1]
```diff
@@ -461,6 +461,13 @@ QtObject:
       self.messageList[chatId].delete
       self.messageList.del(chatId)
 
+  proc toggleReaction*(self: ChatsView, messageId: string, emojiId: int) {.slot.} =
+    if self.activeChannel.id == status_utils.getTimelineChatId():
+      let message = self.messageList[status_utils.getTimelineChatId()].getMessageById(messageId)
+      self.reactions.toggle(messageId, message.chatId, emojiId)
```
Comment (PascalPrecht) [inline src/app/chat/view.nim:467]: Here we determine the "real" chat id of the message from which the reaction is coming from. This is needed because otherwise the reaction is sent to a wrong chat.

File: src/app/chat/views/reactions.nim [hunk 2]
```diff
@@ -32,6 +33,11 @@ QtObject:
     self.pubKey = status_settings.getSetting[string](Setting.PublicKey, "0x0")
 
   proc messageEmojiReactionId(self: ReactionView, chatId: string, messageId: string, emojiId: int): string =
+    let chat = self.status.chat.channels[chatId]
+    var chatId = chatId
+    if chat.chatType == ChatType.Profile:
+      chatId = status_utils.getTimelineChatId()
```
Comment (PascalPrecht) [inline src/app/chat/views/reactions.nim:39]: When reactions are removed, we need to ensure we're removing them from the timeline message list.

File: src/app/chat/views/reactions.nim [hunk 3]
```diff
@@ -42,17 +48,20 @@ QtObject:
...
       self.status.chat.removeEmojiReaction(emojiReactionId)
     
   proc push*(self: ReactionView, reactions: var seq[Reaction]) =
     let t = reactions.len
     for reaction in reactions.mitems:
-      let messageList = self.messageList[][reaction.chatId]
+      let chat = self.status.chat.channels[reaction.chatId]
+      var messageList = self.messageList[][reaction.chatId]
+      if chat.chatType == ChatType.Profile:
+        messageList = self.messageList[][status_utils.getTimelineChatId()]
```
Comment (PascalPrecht) [inline src/app/chat/views/reactions.nim:64]: Same way, when new reactions are coming in, we need to put them in the timeline message list, in case they are coming from a profile update

File: ui/app/AppMain.qml [hunk 4]
```diff
@@ -34,6 +34,19 @@ RowLayout {
         return popup
     }
 
+    function getUserNickname(pubKey) {
```
Comment (jrainville) [inline ui/app/AppMain.qml:37]: getUserNickname is also used in CommunityMembersPopup and GroupInfoPopup, you'll need to point to appMain there too

**Review by jrainville — CHANGES_REQUESTED** [review]
Looks good, just need to fix the `getUserNickname` link in other files too

File: ui/app/AppMain.qml (hunk 4 above)
Comment (PascalPrecht) [reply ui/app/AppMain.qml:37]: Good catch!!!

**Review by jrainville — APPROVED** [review]
